### PR TITLE
Add solutions and test files for Palindrome number and Remove Duplica…

### PR DESCRIPTION
**Pull request by njeridorcaswanja**
…tes from Sorted Array

---
name: solution review 
about: A template PR for code review with a checklist
---

# Behavior

## Files

- [ ] The file name describes the function's behavior
- [ ] There is a module docstring in the function file
- [ ] The test file's name matches the function file name -
  `/tests/test_file_name.py`
- [ ] There is a module docstring in the tests file

### Unit Tests

- [ ] The test class has a helpful name in PascalCase
- [ ] The test class has a docstring
- Every unit test has
  - [ ] A helpful name
  - [ ] A clear docstring
  - [ ] Only one assertion
  - [ ] There is no logic in the unit test
- [ ] All tests pass
- [ ] There are tests for defensive assertions
- [ ] There are tests for boundary cases

### Function Docstring

- [ ] The function's behavior is described
- The function's arguments are described:
  - [ ] Type
  - [ ] Purpose
  - [ ] Other assumptions (eg. if it's a number, what's the expected range?)
- The return value is described
  - [ ] Type
  - [ ] Other assumptions are documented
- The defensive assertions are documented using `Raises:`
  - [ ] Each assumption about an argument is checked with an assertion
  - [ ] Each assertion checks for _only one_ assumption about the argument
- [ ] Include 3 or more (passing!) doctests

### The Function

- [ ] The function's name describes it's behavior
- [ ] The function's name matches the file name
- [ ] The function has correct type annotations
- [ ] The function is not called in the function file

### Do's

- [ ] Variable names help to understand the strategy
- [ ] Any comments are clear and describe the strategy
- [ ] Lines of code are spaced to help show different stages of the strategy

### Don'ts

- [ ] The function's strategy _is not_ described in the documentation
- [ ] Comments explain the _strategy_, **not** the _implementation_
- [ ] The function _does not_ have more comments than code
  - If it does, consider finding a new strategy or a simpler implementation

## Implementation

- [ ] The code passes the formatting checks
- [ ] The code passes all Ruff linting checks
- [ ] The code has no (reasonable) Pylint errors
  - In code review, you can decide when fixing a Pylint error is helpful and
    when it's too restricting.
- [ ] Variables are named with snake_case
- [ ] Variable names are clear and helpful
- [ ] The code follows the strategy as simply as possible
- [ ] The implementation is as simple as possible given the strategy
- [ ] There are no commented lines of code
- [ ] There are no `print` statements anywhere
- [ ] The code includes defensive assertions
- [ ] Defensive assertions include as little logic as possible

### CODE REVIEW

File: solutions/palindrome number.py [hunk 1]
```diff
@@ -0,0 +1,49 @@
+#!/usr/bin/env python3
+# -*- coding: utf-8 -*-
+"""
+Created on 11/01/2025
+@author: Dorcas Wanja Njeri
+"""
```
Comment (Fikre-Mamo) [inline solutions/palindrome number.py]: add docstring for the unittest

File: solutions/tests/test_Remove_Duplicates_from_sorted_Array.py [hunk 2]
```diff
@@ -0,0 +1,64 @@
...
+# -*- coding: utf-8 -*-
+"""
+Created on 11/01/2025
+@author: Dorcas Wanja Njeri
+"""
+
+import unittest
+
+from solutions import remove_duplicates_from_sorted_array
+
```
Comment (Fikre-Mamo) [inline solutions/tests/test_Remove_Duplicates_from_sorted_Array.py]: changed to <solutions.your code name>

File: solutions/tests/test_Remove_Duplicates_from_sorted_Array.py [hunk 3]
```diff
@@ -0,0 +1,64 @@
...
+        self.assertEqual(length, 0)
+
+    def test_single_element(self):
+        """Should return the same list if there is only one element"""
+        nums = [5]
+        length = remove_duplicates_from_sorted_array(nums)
+        expected = [5]
+        self.assertEqual(length, 1)
+        self.assertEqual(nums[:length], expected)
+
```
Comment (Fikre-Mamo) [inline solutions/tests/test_Remove_Duplicates_from_sorted_Array.py]: one assert for each test

**Review by Fikre-Mamo — CHANGES_REQUESTED** [review]

**Comment by Fikre-Mamo** [comment]
Screenshot 2025-01-11 074412
# install ruff
`pip install ruff`
### after that run:
`ruff format .`
### then run:
`ruff check`


## What is ruff?
ruff is tool in python used to format the code in proper way!

**Comment by njeridorcaswanja** [comment]
Thank you I have been trying to do that but I am unable to

**Comment by Fikre-Mamo** [comment]
if ruff doesn't work tray WSL

**Comment by Fikre-Mamo** [comment]
Install WSL on your pc then go to Microsoft store and install ubuntu

**Comment by njeridorcaswanja** [comment]
I managed to use ruff to format the code thank you

**Comment by Fikre-Mamo** [comment]
## the error still rise
image

**Comment by Fikre-Mamo** [comment]
## Did you run ruff like this?
Screenshot 2025-01-11 204733
Screenshot 2025-01-11 204807

**Comment by Fikre-Mamo** [comment]
## And insure to turn on the auto save first
Screenshot 2025-01-11 205443

File: solutions/palindrome number.py (hunk 1 above)
Comment (njeridorcaswanja) [reply solutions/palindrome number.py]: done

File: solutions/tests/test_Remove_Duplicates_from_sorted_Array.py (hunk 2 above)
Comment (njeridorcaswanja) [reply solutions/tests/test_Remove_Duplicates_from_sorted_Array.py]: done

File: solutions/tests/test_Remove_Duplicates_from_sorted_Array.py (hunk 3 above)
Comment (njeridorcaswanja) [reply solutions/tests/test_Remove_Duplicates_from_sorted_Array.py]: done

**Review by KimaciaJnr — COMMENTED** [review]
Fix the importation errors and the ruff installation to pass CI checks.

**Comment by njeridorcaswanja** [comment]
I did that 
palindrome Screenshot 2025-01-12 114706

**Comment by njeridorcaswanja** [comment]
and this

**Comment by Fikre-Mamo** [comment]
## re push it again

**Review by Fikre-Mamo — APPROVED** [review]
LGTM!
